File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/infrastructure/PostJpaRepository.java
package com.eager.questioncloud.core.domain.post.infrastructure;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface PostJpaRepository extends JpaRepository<PostEntity, Long> {
    Optional<PostEntity> findByIdAndWriterId(Long id, Long userId);
}

File: qc-core/qc-post/qc-post-core/src/main/kotlin/com/eager/questioncloud/post/infrastructure/repository/PostJpaRepository.kt
package com.eager.questioncloud.post.infrastructure.repository

import com.eager.questioncloud.post.infrastructure.entity.PostEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface PostJpaRepository : JpaRepository<PostEntity, Long> {
    fun findByIdAndWriterId(id: Long, userId: Long): Optional<PostEntity>
}
